airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -918,7 +912,9 @@ def list_processing_jobs(self, **kwargs) -> List[Dict]:  # noqa: D402
         )
         return results
 
-    def _list_request(self, partial_func, result_key: str, max_results: Optional[int] = None) -> List[Dict]:
+    def _list_request(
+        self, partial_func: Callable, result_key: str, max_results: Optional[int] = None
+    ) -> List[Dict]:
```

Comment:
```suggestion
    ) -> List[dict]:
```
if we are not specifying types for keys and values then let's keep it simple with `List[dict]`